### PR TITLE
Finish test for thuai6_cpp_compiler

### PR DESCRIPTION
**Pull request by zzdhybthu**
Contribute:
1. A complete reconstruction of code related routes has been implemented.
2. A simple but complete test was done using code from last year's THUAI6.
3. THUAI_CPP image has been updated to latest version in eesast server.

Note:
1. BASE_DIR environment variable is added, which takes effect only in non-production environments.

TODO:
1. CODE_ID and LANG environment variable passed to the compiled container is underutilized. Update compile.sh in THUAI6 repository for better performance.
2. Code robustness may need further improvement. For example, any 500 error may cause confusion in file structure and database records, which is unexpected.

### CODE REVIEW

File: src/routes/code.ts
Comment (xiangmy21) [inline src/routes/code.ts:1]: 太帅了！帅的惊天动地！

**Review by xiangmy21 — CHANGES_REQUESTED** [review]

File: src/routes/code.ts [hunk 1]
```diff
@@ -191,282 +236,341 @@ router.post("/compile", async (req, res) => {
...
+              }
+            });
+          });
+        };
+        if (compile_status === "Success") {
+          const key = `${contest_name}/code/${team_id}/${code_id}`;
+          const localFilePath = `${utils.base_directory}/${key}`;
+          await uploadObject(localFilePath, key);
+        }
+        const key = `${contest_name}/code/${team_id}/${code_id}.txt`;
```
[the remark this answers is not in the document]
Comment (FranGuam) [reply src/routes/code.ts:405]: 我后来想了想，`THUAI6`的`log`还在吗？如果在的话是不是都是`txt`后缀？如果考虑向前兼容的问题，好像不改比较轻松

[the remark this answers is not in the document]
Comment (FranGuam) [reply src/routes/code.ts:405]: 同时这个问题影响不大（

File: src/routes/code.ts
Comment (FranGuam) [reply src/routes/code.ts:1]: \神/

File: src/routes/code.ts [hunk 2]
```diff
@@ -2,175 +2,220 @@ import express from "express";
...
-          }
-        }
+ * PUT upload code, for test only
+ * @param {string} contest_name
+ * @param {uuid} code_id
+ * @param {uuid} team_id
+ * @param {string} suffix
+ * @param {string} path
+*/
+router.put("/upload", async (req, res) => {
```
[the remark this answers is not in the document]
Comment (zzdhybthu) [reply src/routes/code.ts:64]: 好的。

File: src/routes/code.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (zzdhybthu) [reply src/routes/code.ts:405]: 得让队式那边也改一下，yxgg的最新版镜像还是用的txt

[the remark this answers is not in the document]
Comment (xiangmy21) [reply src/routes/code.ts:405]: 就别想着向前兼容了，之前是这样的（
image
而且文件结构也对不上，不如后面整个脚本处理一下

[the remark this answers is not in the document]
Comment (FranGuam) [reply src/routes/code.ts:405]: > 就别想着向前兼容了，之前是这样的（ 

okay，那就改回`.log`吧，跟李圣说一下，正好满足一下强迫欲

**Review by xiangmy21 — APPROVED** [review]

**Pull request merged**